RedfishPkg/RedfishCredentialDxe/RedfishCredentialDxe.c

```diff
@@ -123,6 +132,668 @@ RedfishCredentialStopService (
   return LibStopRedfishService (This, ServiceStopType);
 }
 
+/**
+  Retrieve platform's Redfish authentication information.
+
+  This functions returns the Redfish authentication method together with the user Id and
+  password.
+  - For AuthMethodNone, the UserId and Password could be used for HTTP header authentication
+    as defined by RFC7235.
+  - For AuthMethodRedfishSession, the UserId and Password could be used for Redfish
+    session login as defined by  Redfish API specification (DSP0266).
+
+  Callers are responsible for and freeing the returned string storage.
+
+  @param[in]   This                Pointer to EDKII_REDFISH_CREDENTIAL2_PROTOCOL instance.
+  @param[out]  AuthMethod          Type of Redfish authentication method.
+  @param[out]  UserId              The pointer to store the returned UserId string.
+  @param[out]  Password            The pointer to store the returned Password string.
+
+  @retval EFI_SUCCESS              Get the authentication information successfully.
+  @retval EFI_ACCESS_DENIED        SecureBoot is disabled after EndOfDxe.
+  @retval EFI_INVALID_PARAMETER    This or AuthMethod or UserId or Password is NULL.
+  @retval EFI_OUT_OF_RESOURCES     There are not enough memory resources.
+  @retval EFI_UNSUPPORTED          Unsupported authentication method is found.
+
+**/
+EFI_STATUS
+EFIAPI
+RedfishCredential2GetAuthInfo (
+  IN  EDKII_REDFISH_CREDENTIAL2_PROTOCOL  *This,
+  OUT EDKII_REDFISH_AUTH_METHOD           *AuthMethod,
+  OUT CHAR8                               **UserId,
+  OUT CHAR8                               **Password
+  )
+{
+  EFI_STATUS  Status;
+
+  if ((AuthMethod == NULL) || (UserId == NULL) || (Password == NULL)) {
+    return EFI_INVALID_PARAMETER;
+  }
+
+  if (mCredentialPrivate == NULL) {
+    DEBUG ((DEBUG_ERROR, "%a: failed with error - %r\n", __func__, EFI_NOT_STARTED));
+    return EFI_NOT_STARTED;
+  }
+
+  Status = mCredentialPrivate->RedfishCredentialProtocol.GetAuthInfo (
+                                                           &mCredentialPrivate->RedfishCredentialProtocol,
+                                                           AuthMethod,
+                                                           UserId,
+                                                           Password
+                                                           );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "%a: Failed to retrieve Redfish credential - %r\n", __func__, Status));
+    return Status;

```

Comment:
we can remove this line, this probably inherits from the previous source code.

Reply:
Agree. Removed. Thank you.